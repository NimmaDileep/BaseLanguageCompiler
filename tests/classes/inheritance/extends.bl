class Animal {
    var age: Int;
    var species: String;
}

class Pet: Animal {
    var name: String;

    fun describe() -> String {
        var desc: String = concat("My name is ", concat(self.name, concat(" and I am ", str(self.age))));
        return desc;
    }
}

var p: Pet = Pet(5, "Dog", "Fido");

print(p.describe());
