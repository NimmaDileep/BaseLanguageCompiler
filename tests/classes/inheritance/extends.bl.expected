My name is Fido and I am 5
